datasetinsights/storage/checkpoint.py

```diff
@@ -5,89 +5,109 @@
 import tempfile
 
 import datasetinsights.constants as const
+from datasetinsights.data.download import download_file
 from datasetinsights.torch_distributed import is_master
 
 from .gcs import GCSClient, gcs_bucket_and_path
 
 logger = logging.getLogger(__name__)
 
 DEFAULT_SUFFIX = "estimator"
-GCS_BASE_STR = "gs://"
 
 
-def checkpoint_file_on_gcs(config):
+class EstimatorCheckpoint:
     """
-    determine whether or not the checkpoint file to load an estimator is on
-    gcs
+    For loading and saving estimator checkpoints
```

Comment:
Please follow milti-line docstring in [pep-257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings). Also, since this is the main interface that users will interact with, it deserves better docstrings describing different behaviors of this class.